### PR TITLE
Collect Movie Assets

### PR DESCRIPTION
**Pull request by iwootten**
Modify the "Collect Assets" to work for movies. This involved modifying the UI, so it was possible to select the correct component, instead of assuming that a "main" component existed.

Modify the strip version modifier to allow for stripping of filenames, to prevent verbose playblast names which where named against their layers.

Additionally, break out the common logic between the quicktime/collect_assets to a shared class hook_utils. This requires addition of the ftrack-hooks to the pythonpath.

screenshot_29
screenshot_30

### CODE REVIEW

File: pipeline_plugins/resource/hook/collect_assets.py [hunk 1]
```diff
@@ -122,35 +133,67 @@ def launch(event):
...
+                        "label": "Strip Version",
+                        "value": "strip_version"
+                    },
+                    {
+                        "label": "Strip Filename",
+                        "value": "strip_filename"
+                    },
+                    {
+                        "label": "Original Filename",
+                        "value": "False"
```
Comment (tokejepsen) [inline pipeline_plugins/resource/hook/collect_assets.py]: Seems a bit odd not to follow the rest of the values with ```original_filename```.

**Comment by tokejepsen** [comment]
Quicktime action is working :)

I tried running collect_assets on multiple assetversions and it copied the same component (same file) multiple times, but with different naming as destination.

When trying to strip the version with collect_assets, and the component does not have a version in the name, it errors. Do you think it should be more fail safe, so try to strip version from the name and if none is found, just continue?

The collection of components to display can become quite large, meaning there will be an unexpected waiting time for the user. Maybe we can run a job for the collection, so at least something visual is happening in the Ftrack UI?

Maybe give the enumerator a default value by adding ```"value": "strip_version"```?

Lastly I think this a good opportunity to reorganize these plugins to separate folders so people can opt in which plugin to use. Then we can also have some more documentation on the plugins with a README similar to; https://github.com/tokejepsen/ftrack-hooks/tree/master/djv_plugin.
Also if we are going to build a shared set of utilities, we should make it into a module called "ftrack_hooks" within the repository. In each plugin that uses the shared utilities, we would document it in the README.

File: pipeline_plugins/resource/hook/collect_assets.py (hunk 1 above)
Comment (iwootten) [reply pipeline_plugins/resource/hook/collect_assets.py]: Hmm, yeah - this was as it was originally true/false. Will update.

**Comment by iwootten** [comment]
> I tried running collect_assets on multiple assetversions and it copied the same component (same file) multiple times, but with different naming as destination.

I hadn't actually realised the plugin could be run with multiple assets selected. Will update to fix that.

With the version stripping, this is the same behaviour as originally, but yes it should be written to handle no version name - seems like the original version stripping code is ripped from the foundry's site.

I think the default should be to use the original filename, given that's effectively non-destructive.

Reorganizing the plugins is really outside the scope of this issue - should be done as part of another PR.

**Comment by tokejepsen** [comment]
> I think the default should be to use the original filename, given that's effectively non-destructive.

Sounds good to me :)

> Reorganizing the plugins is really outside the scope of this issue - should be done as part of another PR.

That is fair enough, but could you make the ftrack_hooks module since you introducing this code?

**Comment by tokejepsen** [comment]
This looks good :)

There is edge case where the user can select "Strip Filename" and no parents, which results in a blank named file. Don't mind merging this anyways, if you are happy with it?

**Comment by iwootten** [comment]
Yeah, think that's ok - if they want to strip the filename, then don't select anything to replace it with, it'll end up with nothing.

**Pull request merged**